### PR TITLE
Feat: Add new design for search results page(no results found)

### LINKED ISSUE #2175: feat: add new design for search results page (no results found)
> You can view the figma file [here](https://www.figma.com/file/ntisIxc9QHwOZ828CH5G3a/LinksHub?type=design&node-id=17-231&mode=design&t=29U3Qw0LktnVRteR-0).

image

> The design should be responsive. 

---

ℹ️ We recommend updating the styling of existing components, otherwise you can add extra components if there is a need.

- You need to implement the above design. 

> Any changes to the above design are welcome. Please feel free to share your thoughts, and we will discuss them thoroughly.

---

### PR DESCRIPTION
**Pull request by heysolomon**
## Fixes Issue
This PR closes #2175

## Changes proposed
- added the new design for the search results page(no results found)

## Screenshots

image

## Note to reviewers
- made an update to the `tailwind.config.js` file by adding new colors that were used for building the **Search page**

### CODE REVIEW

**Review by rupali-codes — COMMENTED** [review]
hey @heysolomon, please make the following changes
- Remove extra space in sidebar,
- Shift the heading to left
- Reduce vertical adding in `join discord` button
- Increase side of the arrow in `join discord` button

Thanks for contributing :)

**Comment by heysolomon** [comment]
> hey @heysolomon, please make the following changes
> 
> * Remove extra space in sidebar,
> * Shift the heading to left
> * Reduce vertical adding in `join discord` button
> * Increase side of the arrow in `join discord` button
> 
> Thanks for contributing :)

Okay, making the fixes

**Comment by heysolomon** [comment]
> hey @heysolomon, please make the following changes
> 
> * Remove extra space in sidebar,
> * Shift the heading to left
> * Reduce vertical adding in `join discord` button
> * Increase side of the arrow in `join discord` button
> 
> Thanks for contributing :)

@rupali-codes I have made the changes
-  [x] Remove extra space in sidebar,
-  [x] Shift the heading to left
-  [x] Reduce vertical adding in `join discord` button
-  [x] Increase size of the arrow in `join discord` button

# Note to reviewers
- I noticed the styles weren't taking effect when I tried reducing the vertical padding of the `Button` component in the `className.` I didn't want to change the padding in the component because I didn't know if the `Button` component was used elsewhere in the code. Ideally, my new styles are supposed to override the old ones. So, I added a tailwind package, [tailwind-merge](https://www.npmjs.com/package/tailwind-merge) a utility function to merge [Tailwind CSS](https://tailwindcss.com/) classes efficiently  in JS without style conflicts.

**Comment by Anmol-Baranwal** [comment]
> > hey @heysolomon, please make the following changes
> > 
> > * Remove extra space in sidebar,
> > * Shift the heading to left
> > * Reduce vertical adding in `join discord` button
> > * Increase side of the arrow in `join discord` button
> > 
> > Thanks for contributing :)
> 
> @rupali-codes I have made the changes
> 
> * [x]  Remove extra space in sidebar,
> * [x]  Shift the heading to left
> * [x]  Reduce vertical adding in `join discord` button
> * [x]  Increase size of the arrow in `join discord` button
> 
> # Note to reviewers
> * I noticed the styles weren't taking effect when I tried reducing the vertical padding of the `Button` component in the `className.` I didn't want to change the padding in the component because I didn't know if the `Button` component was used elsewhere in the code. Ideally, my new styles are supposed to override the old ones. So, I added a tailwind package, [tailwind-merge](https://www.npmjs.com/package/tailwind-merge) a utility function to merge [Tailwind CSS](https://tailwindcss.com/) classes efficiently  in JS without style conflicts.

Yeah, tailwind merge is very useful and addition will just benefit in the longer run.

**Comment by heysolomon** [comment]
> > > hey @heysolomon, please make the following changes
> > > 
> > > * Remove extra space in sidebar,
> > > * Shift the heading to left
> > > * Reduce vertical adding in `join discord` button
> > > * Increase side of the arrow in `join discord` button
> > > 
> > > Thanks for contributing :)
> > 
> > 
> > @rupali-codes I have made the changes
> > 
> > * [x]   Remove extra space in sidebar,
> > * [x]   Shift the heading to left
> > * [x]   Reduce vertical adding in `join discord` button
> > * [x]   Increase size of the arrow in `join discord` button
> > 
> > # Note to reviewers
> > 
> > * I noticed the styles weren't taking effect when I tried reducing the vertical padding of the `Button` component in the `className.` I didn't want to change the padding in the component because I didn't know if the `Button` component was used elsewhere in the code. Ideally, my new styles are supposed to override the old ones. So, I added a tailwind package, [tailwind-merge](https://www.npmjs.com/package/tailwind-merge) a utility function to merge [Tailwind CSS](https://tailwindcss.com/) classes efficiently  in JS without style conflicts.
> 
> Yeah, tailwind merge is very useful and addition will just benefit in the longer run.

Yeah

File: tailwind.config.js [hunk 1]
```diff
@@ -11,12 +11,18 @@ module.exports = {
...
         'dark-primary': '#101623',
         'gray-text': '#9ca3af',
         'text-primary': '#d1d5db',
         'text-secondary': '#4b5563',
+        light: {
+          primary: '#f5f3ff',
+          silver: '#EDEDED',
+        },
+        primary: {
+          purple: '#714EFF',
```
Comment (aftabrehan) [inline tailwind.config.js:24]: The same color `#714EFF`  is already defined as `primary` above. I guess we don't need to redefine it.

File: components/Button/Button.tsx [hunk 2]
```diff
@@ -20,13 +20,13 @@ const Button: FC<Props> = ({
   className,
 }) => {
   const styles = {
-    primary: '', // @todo: add styles when using these button types in ResourcesCards
+    primary: 'bg-primary-purple hover:bg-primary-purple/80 text-white rounded-[10px] duration-500', // @todo: add styles when using these button types in ResourcesCards
```
Comment (aftabrehan) [inline components/Button/Button.tsx:23]: We can remove the comment now.

```suggestion
    primary: 'bg-primary-purple hover:bg-primary-purple/80 text-white rounded-[10px] duration-500',
```

File: components/Button/Button.tsx [hunk 3]
```diff
@@ -20,13 +20,13 @@ const Button: FC<Props> = ({
...
     secondary: '',
     pale: 'bg-slate-100 hover:bg-violet-50 dark:bg-white dark:bg-opacity-10 dark:hover:bg-opacity-20 text-slate-500 dark:text-violet-50',
     text: 'text-slate-400 dark:text-gray-400 !gap-4',
   }[variant]
 
-  const cls = clsx(
+  const cls = cn(
```
Comment (aftabrehan) [inline components/Button/Button.tsx:29]: Any specific reason for using `cn` instead of clsx?

File: components/NotFound/NotFound.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import { FC } from 'react'
+import { IoIosArrowRoundForward } from "react-icons/io";
+import Button from 'components/Button';
+import Search from "assets/icons/svg/search-icon.svg"
+
+const NotFound: FC = () => {
+  return (
+    <div className="pt-6 md:p-0 text-xl w-full my-auto h-[calc(100vh-450px)] md:h-[calc(100vh-270px)]">
```
Comment (aftabrehan) [inline components/NotFound/NotFound.tsx:8]: A few adjustments are required for the mobile view.

Screenshot 2023-12-30 at 8 14 28 PM

Maybe you can utilize `min-height` to fix this. 

```suggestion
    <div className="pt-6 md:p-0 text-xl w-full my-auto min-h-[calc(100vh-450px)] md:min-h-[calc(100vh-270px)]">
```

**Review by aftabrehan — CHANGES_REQUESTED** [review]
Excellent PR, @heysolomon 💪 🚀 💯 

It just needs minor adjustments and then we're good to go 🚀

**Review by rupali-codes — CHANGES_REQUESTED** [review]
@heysolomon can you reduce the space between the search icon the top-heading

image

File: components/Button/Button.tsx (hunk 3 above)
Comment (heysolomon) [reply components/Button/Button.tsx:29]: Yes.

Unlike `clsx`, `cn` is a utility function that merges both `clsx` and `tailwind-merge`. So, to ensure that the function isn't always written/repeated when writing new components, I wrote the function `cn` for this purpose, and I added it to `lib/utils.ts`, which is also a common pattern I see that is used in projects. A UI library like `Shadcn ` also uses the same pattern.

**Comment by heysolomon** [comment]
> @heysolomon can you reduce the space between the search icon the top-heading
> 
> image

Sorry I don't clearly understand.

**Review by rupali-codes — COMMENTED** [review]
@heysolomon i'm talking about this

image

- reduce size of the icon
- put some space around the icon
- make the button like earlier (voilet)

**Comment by heysolomon** [comment]
> @heysolomon i'm talking about this
> 
> image
> 
> * reduce size of the icon
> * put some space around the icon
> * make the button like earlier (voilet)

Oh! okay. Working on it

**Comment by heysolomon** [comment]
@rupali-codes and @Anmol-Baranwal, I fixed the icon size and spacing. 

Here are screenshots of different views
image
Screenshot 2024-01-04 122725
Screenshot 2024-01-04 122744


Please let me know if there's any other thing I need to fix

**Comment by Anmol-Baranwal** [comment]
@rupali-codes 
Please check.

**Review by rupali-codes — APPROVED** [review]
looks good to me

**Review by aftabrehan — APPROVED** [review]
Excellent PR, @heysolomon 💪 🚀 💯

**Comment by aftabrehan** [comment]
@Anmol-Baranwal and @CBID2 please help review and merge this PR. Thank you!

**Review by CBID2 — APPROVED** [review]
Looks good to me too

**Pull request merged**